tests/test_recursion_depth.py

```
@@ -0,0 +1,68 @@
+
+
+def _nest(n):
+    node = "leaf"
+    for _ in range(n):
+        node = {"k": node}
+    return node
+
+
+# Calibrated between the pre-fix reach (~57 on CPython 3.x) and the post-fix reach (~75), ~8 levels
+# of margin each side, and comfortably above the editor codec's deepest self-`include` target
+# (G_encode, depth 41). Re-confirm on the CI matrix (3.9–3.13) when landing.
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)